catalog/transformer/catalog-transformer-service-atom/src/main/java/ddf/catalog/transformer/response/query/atom/AtomTransformer.java

```diff
@@ -473,4 +478,24 @@ private Link addLinkHelper(Action action, Entry entry, String linkType, String m
     }
     return georssPositions;
   }
+
+  private int getCount(SourceResponse queryResponse) {
+    if (queryResponse == null) {
+      return -1;
+    }
+
+    QueryRequest queryRequest = queryResponse.getRequest();
+
+    if (queryRequest == null) {
+      return -1;
+    }
+
+    Serializable countSerializable = queryRequest.getProperties().get("count");
+
+    if (countSerializable == null) {
+      return -1;
```

Comment:
This method returns -1 in case of error. This method's current usage indicates that if the count cannot be obtained, the results should not be treated as a hit count only query, but returning -1 makes this code unclear if this is the intended behavior. I suggest changing the return type of this method to `Optional<Integer>` and returning `Optional.empty()` for error cases to make the intention clearer.

Reply:
-1 was chosen to emulate the current behavior of this code in cases where an invalid count is supplied. I'd like the use of `Optional` here if this were a public method, but I think switching that would increase the complexity of the calling code. I've added a constant to hopefully make the intent more clear.

Reply:
As far as I can tell, `getCount` is only called in one place and that one place just checks that the count `!= 0`. This makes the fact that an invalid count should add all results hidden. With an `Optional`, it is clear that the error is acknowledged and the logic is not much longer:
```
if (getCount(sourceResponse).orElse(-1) != 0) {
```
The calling code is barely more complex, but clearer.

A constant for -1 clarifies the `getCount` method, but since that constant is not used in the `!= 0` check, it still looks suspicious. However, with the constant declaration, I would be comfortable making this :pencil2:.